lib/engine/game/g_1846.rb

```diff
@@ -311,9 +311,7 @@ def action_processed(action)
 
         check_special_tile_lay(action)
 
-        @corporations.dup.each do |corporation|
-          close_corporation(corporation) if corporation.share_price&.price&.zero?
-        end
+        super
```

Comment:
why is this calling super

Reply:
shouldn't this call @corporations.dup close corporation?

Reply:
i see, you changed action_processed to close corporations... that should not be done, it's going to happen on every single action

Reply:
Why? Doesn't it get called exactly in the cases? The point was to generalise this for all games that have a close corp space on the stock market.
Maybe I should put a condition surrounding the `each`?

Reply:
because after every single action, you're going to duplication corporations and iterate over them

Reply:
But that's not something new I did. I just moved the code from 1846 to the main method.

What do you propose I do?

Reply:
so are you saying this code doesn't run in 1889?

Reply:
I was not understanding if you wanted to change how this is done altogether or just check if we're in a game where a corp can close.

Now it doesn't run in 1889.